dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackedEntityAttributeValueDeletionHandler.java

```diff
@@ -40,21 +39,19 @@
  */
 @Component
 @AllArgsConstructor
-public class TrackedEntityAttributeValueDeletionHandler extends DeletionHandler {
+public class TrackedEntityAttributeValueDeletionHandler extends JdbcDeletionHandler {
   private static final DeletionVeto VETO =
       new DeletionVeto(
           TrackedEntityAttributeValue.class, "Some values are still assigned to this attribute");
 
-  private final TrackedEntityAttributeValueService attributeValueService;
-
   @Override
   protected void register() {
     whenVetoing(TrackedEntityAttribute.class, this::allowDeleteTrackedEntityAttribute);
   }
 
   private DeletionVeto allowDeleteTrackedEntityAttribute(TrackedEntityAttribute attribute) {
-    return attributeValueService.getCountOfAssignedTrackedEntityAttributeValues(attribute) == 0
-        ? ACCEPT
-        : VETO;
+    String sql =
+        "select 1 from trackedentityattributevalue where trackedentityattributeid = :id limit 1";
```

Comment:
do we need to add "limit 1" if `trackedentityattributeid` is the PK?

Reply:
Yes, this is a one-to-many relationship.
For a `TrackedEntityAttribute` we can have multiple `TrackedEntityAttributeValue`s, and we just want to check that it exists